core/src/HIP/Kokkos_HIP_Parallel_Range.hpp

```diff
@@ -108,7 +108,11 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
   inline void execute() const {
     const typename Policy::index_type nwork = m_policy.end() - m_policy.begin();
 
-    const int block_size = 256;  // FIXME_HIP Choose block_size better
+    const int block_size =
+        LaunchBounds::maxTperB
+            ? LaunchBounds::maxTperB
+            : ::Kokkos::Experimental::Impl::HIPTraits::
+                  MaxThreadsPerBlock;  // FIXME_HIP Choose block_size better
```

Comment:
This seems to be on purpose. Is 1024 better?

Reply:
Moved this bit over to #3299